### PR TITLE
Update the closed record syntax

### LINKED ISSUE #14113: Update closed record syntax
From 2019r1 Spec, The Closed record syntax has been changed. 

* [ ] The syntax for closed record types now uses {| and |} delimiters rather than !....
* [ ] Record binding patterns now also use {| and |} delimiters instead of !....

### PR DESCRIPTION
**Pull request by pubudu91**
```ballerina
type Person record {|
    string name;
    int age;
|}
```

### CODE REVIEW

File: compiler/ballerina-lang/src/main/resources/grammar/BallerinaParser.g4 [hunk 1]
```diff
@@ -168,13 +167,27 @@ typeName
...
+closedRecordTypeDescriptor
+    :   RECORD LEFT_BRACE PIPE fieldDescriptor* PIPE RIGHT_BRACE
+    ;
+
+fieldDescriptor
+    :   fieldDefinition
+    |   typeReference
+    ;
+
+//recordFieldDefinitionList
```
Comment (hasithaa) [inline compiler/ballerina-lang/src/main/resources/grammar/BallerinaParser.g4]: Remove comment?

**Comment by hasithaa** [comment]
@pubudu538 Shall we update the match pattern with the same PR?

**Comment by hasithaa** [comment]
@pubudu91 @NipunaMarcus Shall we mark this PR as WIP?

**Comment by NipunaMarcus** [comment]
Yeah that will be a good idea as it has some work to do to get the source gen and formatting to work ... +1

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/formatting/FormattingNodeTree.java [hunk 2]
```diff
@@ -2969,6 +2969,56 @@ public void formatRecordTypeNode(JsonObject node) {
...
+                        String text = currentWS.get(FormattingConstants.TEXT).getAsString();
+                        if (text.equals("record")) {
+                            if (formatConfig.get(FormattingConstants.NEW_LINE_COUNT).getAsInt() > 0) {
+                                currentWS.addProperty(FormattingConstants.WS,
+                                        this.getNewLines(formatConfig.get(FormattingConstants.NEW_LINE_COUNT)
+                                                .getAsInt()) + indentation);
+                            } else {
+                                currentWS.addProperty(FormattingConstants.WS, FormattingConstants.SINGLE_SPACE);
+                            }
+                        } else if (text.equals("{")) {
```
Comment (hasithaa) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/formatting/FormattingNodeTree.java]: Better to move these to constants.

Comment (pubudu91) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/formatting/FormattingNodeTree.java]: @NipunaMarcus FYI

Comment (NipunaMarcus) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/formatting/FormattingNodeTree.java]: @hasithaa working on it  as i'm doing a refactoring ....

**Pull request merged**